include/wlr/types/wlr_client_buffer.h

```diff
@@ -13,9 +13,25 @@
 #include <wayland-server-core.h>
 #include <wlr/types/wlr_buffer.h>
 
+struct wlr_renderer;
+struct wlr_client_buffer;
+
+struct wlr_client_buffer_impl {
+	bool (*is_instance)(struct wl_resource *resource);
+	bool (*get_buffer_size)(struct wl_resource *resource,
+			struct wlr_renderer *renderer, int *width, int *height);
+	bool (*initialize)(struct wlr_client_buffer *,
+			struct wlr_renderer *renderer);
+	bool (*apply_damage)(struct wlr_client_buffer *buffer,
+			struct wl_resource *resource, pixman_region32_t *damage);
```

Comment:
Not yet sure it makes sense for this to be part of the interface. Ideally we should be able to rework `wlr_texture_write_pixels` and make it not depend on the client buffer implementation.

Reply:
Yeah atm it's only used for shm buffers, what's your idea for `wlr_texture_write_pixels` ?